src/ucp/wireup/select.c

```diff
@@ -1246,24 +1254,33 @@ static UCS_F_NOINLINE void
 ucp_wireup_select_ctx_init(ucp_wireup_select_ctx_t *select_ctx,
                            ucp_ep_h ep, const ucp_ep_params_t *params,
                            unsigned ep_init_flags, unsigned address_count,
-                           const ucp_address_entry_t *address_list)
+                           const ucp_address_entry_t *address_list,
+                           uint64_t tl_bitmap)
 {
-    select_ctx->ep                 = ep;
-    select_ctx->params             = params;
-    select_ctx->ep_init_flags      = ep_init_flags;
-    select_ctx->address_list       = address_list;
-    select_ctx->address_count      = address_count;
-    select_ctx->num_lanes          = 0;
-    select_ctx->allow_am           =
-        ucp_wireup_allow_am_emulation_layer(params, ep_init_flags);
+    select_ctx->ep            = ep;
+    select_ctx->params        = params;
+    select_ctx->ep_init_flags = ep_init_flags;
+    select_ctx->tl_bitmap     = tl_bitmap;
+    select_ctx->address_list  = address_list;
+    select_ctx->address_count = address_count;
+    select_ctx->num_lanes     = 0;
+    select_ctx->allow_am      = ucp_wireup_allow_am_emulation_layer(params,
+                                                                    ep_init_flags);
+    /* If we are using reduced TL bitmap, don't show errors */
+    select_ctx->show_error    = (tl_bitmap == -1);
+
     memset(select_ctx->lane_descs, 0, sizeof(select_ctx->lane_descs));
+    memset(&select_ctx->am_info, 0, sizeof(select_ctx->am_info));
 }
 
 static UCS_F_NOINLINE ucs_status_t
 ucp_wireup_search_lanes(ucp_wireup_select_ctx_t *select_ctx,
                         const ucp_ep_config_key_t *key)
 {
-    ucs_status_t status;
+    ucs_status_t status, am_status;
+
+    /* Selecting AM will fill AM select info object */
+    am_status = ucp_wireup_select_am_lane(select_ctx, 0);
```

Comment:
maybe am_status should be part of select_ctx?

Reply:
we don't need it anymore - this was needed for the previous implementation only
we can just add AM lane